flang/lib/Semantics/check-omp-structure.cpp

```diff
@@ -4550,6 +4550,95 @@ void OmpStructureChecker::Leave(const parser::DoConstruct &x) {
   Base::Leave(x);
 }
 
+void OmpStructureChecker::Enter(const parser::OpenMPInteropConstruct &x) {
+  bool isTargetSyncOccured{false};
+  bool isDependClauseOccured{false};
+  int targetCount{0}, targetSyncCount{0};
+  const auto &dir{std::get<parser::Verbatim>(x.t)};
+  std::list<std::string> ObjectNameList;
+  PushContextAndClauseSets(dir.source, llvm::omp::Directive::OMPD_interop);
+  const auto &clauseList{std::get<parser::OmpClauseList>(x.t)};
+  for (const auto &clause : clauseList.v) {
+    common::visit(
+      common::visitors{
+        [&](const parser::OmpClause::Init &InitClause) {
+          const auto &InteropTypeList{
+              std::get<parser::OmpInitClause::InteropTypes>(InitClause.v.t)};
+          for (auto &InteropTypeVal : InteropTypeList.v) {
+            if (*(parser::Unwrap<parser::InteropType::Kind>(
+                    InteropTypeVal)) ==
+                parser::InteropType::Kind::TargetSync) {
+              isTargetSyncOccured = true;
+              ++targetSyncCount;
+            } else {
+              ++targetCount;
+            }
+            if (targetCount >= 2 || targetSyncCount >= 2) {
+              context_.Say(GetContext().clauseSource,
+                          "Each interop-type may be speciﬁed "
+                          "at most once."_err_en_US);
+            } 
+          }
+          const auto &InteropVar{parser::Unwrap<parser::OmpObject>(std::get<parser::OmpInitClause::InteropVar>(InitClause.v.t))};
+          const auto *name{parser::Unwrap<parser::Name>(InteropVar)};
+          const auto ObjectName{name->ToString()};
+          if(ObjectNameList.empty()) {
+            ObjectNameList.push_back(ObjectName);
+          } else {
+            if(ObjectNameList.end() != std::find(ObjectNameList.begin(),ObjectNameList.end(),ObjectName)) {
+              context_.Say(GetContext().clauseSource,
+                          "Each interop-var may be speciﬁed for at most one action-clause "
+                          "of each interop construct."_err_en_US);
+            }
+          }
```

Comment:
Or you can use `SmallVectorImpl`. This simplifies the check.

Reply:
Addressed the comment on [#1c57d1e](https://github.com/swatheesh-mcw/llvm-project/pull/1/commits/1c57d1ea2b8e7095a834689ad5b832e6a2a84c37)

Reply:
Usually, it is recommended to use `SmallVectorImpl` or `ArrayRef` here instead of `std::list`

These containers has some useful builtin functions. Like there is something like contrained function to check for element in the list